nh_eobs_mental_health/models/nh_clinical_wardboard.py

```diff
@@ -0,0 +1,91 @@
+from openerp.osv import orm, osv, fields
+
+
+class NHClinicalWardboard(orm.Model):
+
+    _name = 'nh.clinical.wardboard'
+    _inherit = 'nh.clinical.wardboard'
+
+    def _get_obs_stop_from_spell(self, cr, uid, ids, field_name, arg,
+                                 context=None):
+        """
+        Function field to return obs_stop flag from spell
+        :param cr: Odoo cursor
+        :param uid: User ID of user doing operatoin
+        :param ids: Ids to read
+        :param field_name: name of field
+        :param arg: arguments
+        :param context: Odoo context
+        :return: obs_stop flag from spell
+        """
+        spell_model = self.pool['nh.clinical.spell']
+        flags = spell_model.read(cr, uid, ids, ['obs_stop'], context=context)
+        return dict([(rec.get('id'), rec.get('obs_stop')) for rec in flags])
+
+    _columns = {
+        'obs_stop': fields.function(_get_obs_stop_from_spell, type='boolean')
+    }
+
+    def toggle_obs_stop(self, cr, uid, ids, context=None):
+        """
+        Handle button press on 'Stop Observations'/'Restore Observation' button
+        :param cr: Odoo cursor
+        :param uid: User doing the action
+        :param ids: IDs of wardboard
+        :param context: Odoo context
+        :return: True
+        """
+        spell_model = self.pool['nh.clinical.spell']
+        if isinstance(ids, list):
+            ids = ids[0]
+        wardboard_obj = self.read(cr, uid, ids, context=context)
+        escalation_tasks_open_warning = 'One or more escalation tasks for ' \
+                                        '{0} are not completed.'
+        spell_activity_id = wardboard_obj.get('spell_activity_id')[0]
+        patient = wardboard_obj.get('patient_id')
+        patient_name = patient[1]
+        spell_id = spell_model.search(
+            cr, uid, [['patient_id', '=', patient[0]]])
+        if not spell_id:
+            raise ValueError('No spell found for patient')
+        self.toggle_obs_stop_flag(cr, uid, spell_id[0], context=context)
+        if self.spell_has_open_escalation_tasks(cr, uid, spell_activity_id,
+                                                context=context):
+            raise osv.except_osv(
```

Comment:
On potential solution is to change the button from an object to an action. This will then allow for a view to be returned